js/controller.js

```diff
@@ -1163,6 +1165,27 @@ OO.plugin("Html5Skin", function (OO, _, $, W) {
       this.mb.publish(OO.EVENTS.DISCOVERY_API.SEND_DISPLAY_EVENT, eventData);
     },
 
+    toggleVideoQualityPopOver: function() {
+      this.state.videoQualityOptions.showVideoQualityPopover = !this.state.videoQualityOptions.showVideoQualityPopover;
+      this.renderSkin();
+    },
+
+    receiveVideoQualityChangeEvent: function(eventName, targetBitrate) {
+        this.state.videoQualityOptions.selectedBitrate = {
+        "id": targetBitrate
+      };
+      this.renderSkin({
+          "videoQualityOptions": {
+            "availableBitrates": this.state.videoQualityOptions.availableBitrates,
+            "selectedBitrate": this.state.videoQualityOptions.selectedBitrate,
+            "showVideoQualityPopover":this.state.videoQualityOptions.showVideoQualityPopover
+          }
+        });
+      if(this.state.videoQualityOptions.showVideoQualityPopover == true) {
+        this.toggleVideoQualityPopOver();
+      }
```

Comment:
This was some time ago, but from the ticket I have this:

On calling pp.setTargetBitrate(1), the control directly goes to video controller without updating the skin. For this we have to publish a new event for sendVideoQualityChangeEvent() function in the controller.js which will set the new bitrate.
Tested creating an event listener in controller.js and published the event from setTargetBitrate() and also initialized the new event listener in the Constants.js of bit wrapper but getting an error.

So we made a new event because SET_TARGET_BITRATE was not available in this scope.